core/Constants.php

```diff
@@ -229,10 +229,19 @@ function stripslashes_recursively(&$array) {
 		define('BASE_URL', rtrim($dir, DIRECTORY_SEPARATOR));
 	}
 }
-define('MODULES_DIR', 'modules');
+
+if (!defined('MODULES_DIR')) {
+	define('MODULES_DIR', 'modules');
```

Comment:
We have NEVER respected this constant before... I'm surprised to see it actually.